packages/react-ui/src/app/routes/platform/users/update-role-dialog.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button } from "@/components/ui/button";
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle, DialogTrigger } from "@/components/ui/dialog";
+import { Form, FormField, FormItem, FormMessage } from "@/components/ui/form";
+import { Label } from "@/components/ui/label";
+import { Select, SelectContent, SelectGroup, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
+import { INTERNAL_ERROR_TOAST, useToast } from "@/components/ui/use-toast";
+import { platformUserApi } from "@/lib/platform-user-api";
+import { typeboxResolver } from "@hookform/resolvers/typebox";
+import { useMutation } from "@tanstack/react-query";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { PlatformRole, UpdateUserRequestBody } from "@activepieces/shared";
+
+export const UpdateUserRoleDialog = ({ children, onUpdate, userId, role }: { children: React.ReactNode, onUpdate: (role: PlatformRole) => void, userId: string, role: PlatformRole }) => {
+  const [open, setOpen] = useState(false);
+  const form = useForm<{ role: PlatformRole }>({
+    defaultValues: {
+      role,
+    },
+    resolver: typeboxResolver(UpdateUserRequestBody),
+  });
+  const { toast } = useToast();
+  const { mutate, isPending } = useMutation({
+    mutationKey: ['update-user'],
+    mutationFn: async (request: { platformRole: PlatformRole }) => {
+      await platformUserApi.update(userId, request);
+      onUpdate(request.platformRole);
```

Comment:
Nit:
I would move `onUpdate` to `onSuccess` and just make mutationFn: (request) => platformUserApi.update(userId, request)